Syntax checker now populates the gutter

kaiwood opened the pull request:
Currently, the output of the syntax check is shown in a simple balloon popover. With this patch, the new gutter is populated line by line with a warning sign containing the error message.

infininight commented:
Populating the gutter is good but with this change I get a truncated tooltip with multiple errors. Can you change it to still return the full tooltip as well?

sorbits commented:
When calling `mate` then use ``` `"$TM_MATE" …` ``` this delays the environment variable expansion to the shell which is syntactically simpler and technically better because we do not have to worry about shell escaping the value (as long as we double-quote the variable).

kaiwood commented:
@infininight My previous idea was to only show the current error so one can traverse the error messages one by one (Fixing a line, hitting ^V, fixing another...). But you are right, it doesn't hurt to show the complete tooltip as the error for the current line is the on top of the list. Changed it.

@sorbits Makes sense, thanks. I changed it according to your recommendation.

Force-pushed both changes to the PR, thanks for the feedback!

infininight commented:
Added as f9c884e3fc709b2499e6f79d3e13adac24d72c26, thanks!
